src/Regulator.php

```diff
@@ -139,20 +140,29 @@ public function untrack_action(): void {
 	 * Dispatches a task to be processed later.
 	 *
 	 * @since 0.0.1
+	 * @since 0.0.7 Updated to check if the Shepherd tables have been registered already.
+	 * @since 0.0.7 Updated to use the `action_scheduler_init` hook instead of the `init` hook to check if Action Scheduler is initialized.
 	 *
 	 * @param Task $task  The task to dispatch.
 	 * @param int  $delay The delay in seconds before the task is processed.
 	 *
 	 * @return self The regulator instance.
 	 */
 	public function dispatch( Task $task, int $delay = 0 ): self {
-		if ( did_action( 'init' ) || doing_action( 'init' ) ) {
+		$prefix = Config::get_hook_prefix();
+
+		if ( ! did_action( "shepherd_{$prefix}_tables_registered" ) ) {
+			_doing_it_wrong( __METHOD__, esc_html__( 'Shepherd tables are not yet registered.', 'stellarwp-shepherd' ), '0.0.7' );
```

Comment:
This will appear in the error log: is this the inteinted behavior? I understand that the method should bail, but I'm not sure this user-facing error is the best way to inform about the action to take to solve the issue?
What action should be taken to solve the issue?  Can it be automatically fixed?